src/main/java/com/android/volley/toolbox/DiskBasedCacheUtility.java

```diff
@@ -0,0 +1,143 @@
+package com.android.volley.toolbox;
+
+import androidx.annotation.VisibleForTesting;
+import com.android.volley.Header;
+import com.android.volley.toolbox.DiskBasedCache.CountingInputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+class DiskBasedCacheUtility {
+
+    /** Default maximum disk usage in bytes. */
+    static final int DEFAULT_DISK_USAGE_BYTES = 5 * 1024 * 1024;
+
+    /** High water mark percentage for the cache */
+    @VisibleForTesting static final float HYSTERESIS_FACTOR = 0.9f;
+
+    /** Magic number for current version of cache file format. */
+    static final int CACHE_MAGIC = 0x20150306;
```

Comment:
Note that I don't see this being used outside of CacheHeader, so you could consider removing it from here and keep the one in CacheHeader

Reply:
Yeah, that's smart. I'll make that change.